lib/Widgets/Toast.vala

```diff
@@ -16,11 +16,21 @@
  * {{../doc/images/Toast.png}}
  */
 public class Granite.Toast : Gtk.Widget {
+    /**
+     * Reason why a Toast was closed
+     * @since 7.5.0
+     */
+    [Version (since = "7.5.0")]
+    public enum CloseReason {
+        EXPIRED = 1,
+        DISMISSED = 2,
+        WITHDRAWN = 3
+    }
 
     /**
-     * Emitted when the Toast is closed by activating the close button
+     * Emitted when the Toast is closed
      */
-    public signal void closed ();
+    public signal void closed (int reason_id);
```

Comment:
Any reason to use `int` rather than `uint` as this is an `enum`?

Reply:
Better still, the type should probably be `CloseReason`.

```suggestion
    public signal void closed (CloseReason reason);
```